BedrockServer.cpp

```diff
@@ -228,6 +228,27 @@ void BedrockServer::sync(SData& args,
                 // As it's a quorum commit, we'll need to read from peers. Let's start the next loop iteration.
                 continue;
             }
+        } else if ((preUpdateState == SQLiteNode::MASTERING || preUpdateState == SQLiteNode::STANDINGDOWN)
+                   && nodeState == SQLiteNode::SEARCHING) {
+            // If we were MASTERING, but now we're searching, then something's gone wrong (perhaps we got disconnected
+            // from the cluster). We should give up an any commands, and let them be re-escalated. If commands were
+            // initiated locally, we can just re-queue them, they will get re-checked once things clear up, and then
+            // they'll get processed here, or escalated to the new master.
+            // Commands initiated on slaves just get dropped, they will need to be re-escalated, potentially to a
+            // different master.
+            int requeued = 0;
+            int dropped = 0;
+            try {
+                while (true) {
+                    command = syncNodeQueuedCommands.pop();
+                    if (command.initiatingClientID) {
+                        // This one came from a local client, so we can save it for later.
+                        server._commandQueue.push(move(command));
+                    }
+                }
+            } catch (const out_of_range& e) {
+                SWARN("Abruptly stopped MASTERING. Re-queued " << requeued << " commands, Dropped " << dropped << " commands.");
```

Comment:
Do you it this way to avoid getting a lock on both `syncNodeQueuedCommands` and `server._commandQueue` simultaneously?  Is there any problem resulting from the tiny period during which the `command` is in neither of those lists?  (If we got a lock on both we could move it between the lists in a single atomic move, but this doesn't do that.)